storage/src/objects/block_header.rs

```diff
@@ -64,29 +64,35 @@ impl<T: TransactionScheme, P: LoadableMerkleParameters, S: Storage> Ledger<T, P,
         // Start from the latest block and work backwards
         let mut index = self.get_current_block_height();
 
-        // Update the step size with each iteration
-        let mut step = 1;
+        // The number of "chunks" into which the height is to be divided after the quadratic
+        // step used initially grows too high.
+        let divisor = std::cmp::min(16u32, std::cmp::max(index, 1));
```

Comment:
the desired divisor is `16`, but for block heights lower than that, it's the block height or `1` for a block height of `0`